setup.py

```diff
@@ -19,5 +19,6 @@
         "pymysql",
         "requests",
         "xlrd",
+        "pre-commit",
```

Comment:
1) pre-commit is a dev dep so I would recommend the [following change](https://github.com/cornellius-gp/gpytorch/blob/91fb9ecc431399425e8a318accb4e5191c1b069e/setup.py#L59-L60) to the setup.py and adding it to extras/dev

2) Do users need black/isort locally? If so should they be added to the extras/dev deps like precommit

Reply:
1. Pretty slick, will add.
2. Users don't need to install black/isort. Pre-commit pulls those packages from github and installs them in a special pre-commit location.